cloud/test/recycler_test.cpp

```diff
@@ -4051,4 +4055,232 @@ TEST(RecyclerTest, delete_tmp_rowset_without_resource_id) {
     }
 }
 
+static std::string generate_random_string(int length) {
+    std::string char_set = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+    std::random_device rd;
+    std::mt19937 generator(rd());
+    std::uniform_int_distribution<int> distribution(0, char_set.length() - 1);
+
+    std::string randomString;
+    for (int i = 0; i < length; ++i) {
+        randomString += char_set[distribution(generator)];
+    }
+    return randomString;
+}
+
+std::string instance_id = "concurrent_recycle_txn_label_test_" + generate_random_string(10);
+
+int put_single_kv(const std::unique_ptr<Transaction>& txn, int64_t i) {
+    std::string key;
+    std::string val;
+    int64_t db_id = i;
+    int64_t txn_id = 100000 + i;
+    int64_t sub_txn_id = 200000 + i;
+    int64_t current_time = duration_cast<std::chrono::milliseconds>(
+                                   std::chrono::system_clock::now().time_since_epoch())
+                                   .count();
+
+    // RecycleTxnKeyInfo -> RecycleTxnPB
+    RecycleTxnKeyInfo recycle_txn_key_info {instance_id, db_id, txn_id};
+    recycle_txn_key(recycle_txn_key_info, &key);
+    RecycleTxnPB recycle_txn_pb;
+    if (i < 8000) {
+        recycle_txn_pb.set_creation_time(current_time - 4 * 24 * 3600 * 1000L);
+    } else {
+        recycle_txn_pb.set_creation_time(current_time);
+    }
+    recycle_txn_pb.set_label("recycle_txn_key_info_label_" + std::to_string(i));
+    if (!recycle_txn_pb.SerializeToString(&val)) {
+        LOG_WARNING("failed to serialize recycle txn info")
+                .tag("key", hex(key))
+                .tag("db_id", db_id)
+                .tag("txn_id", txn_id);
+        return -1;
+    }
+    LOG(INFO) << fmt::format("instance_id: {}, db_id: {}, label: {}, k: {}", instance_id, db_id,
+                             recycle_txn_pb.label(), hex(key));
+    txn->put(key, val);
+
+    // TxnIndexKey -> TxnIndexPB
+    key.clear();
```

Comment:
better not to reuse var key value
use new var names like
txn_idx_key txn_idx_value